spec/services/publish_draft_edition_service_spec.rb

```diff
@@ -31,15 +37,23 @@
     end
 
     context "when there is a live edition" do
+      let(:document) { create(:document, :with_current_and_live_editions) }
+      let(:current_edition) { document.current_edition }
+
       it "supersedes the live edition" do
-        document = create(:document, :with_current_and_live_editions)
-        current_edition = document.current_edition
         live_edition = document.live_edition
 
         described_class.call(current_edition, user, with_review: true)
         expect(document.live_edition).to eq(current_edition)
         expect(live_edition).to be_superseded
```

Comment:
while you're in here it's a bit weird that we set a variable of live_edition on line 44 and then don't use it on 47. I'd suggest removing line 44 and switch line 48 to use document.live_edition too.

Do we need to set current_edition let? if not I'd suggest we could reference that off document for consistency with document.live_edition

Reply:
The reason we're setting live_edition in a variable beforehand is because after the class is run the live edition changes to the current edition, meaning if we referenced `document.live_edition` in the expectation -`expect(live_edition).to be_superseded` then it would use the newly published edition (current_edition) where as we want to assert the previously published edition if that makes sense?

> Do we need to set current_edition let? if not I'd suggest we could reference that off document for consistency with document.live_edition

Nope, current_edition now references from document!